lib/modules/platform/bitbucket-server/index.ts

```diff
@@ -691,6 +691,43 @@ export async function addReviewers(
   ]);
 }
 
+/**
+ * Resolves Bitbucket users by username, display name, or email address (userinfo string),
+ * restricted to users who have REPO_READ permission on the target repository.
+ *
+ * Uses the Bitbucket Server REST API:
+ * https://developer.atlassian.com/server/bitbucket/rest/v906/api-group-system-maintenance/#api-api-latest-users-get
+ *
+ * Behavior:
+ * - If no users match the filter, returns empty array.
+ * - Returns the `slug` of the matching active users.
+ *
+ * @param userinfo - A string that could be the user's email, display name, or username.
+ * @returns An object containing the user's `slugs` for active found users.
+ */
+async function getUserDetails(userinfo: string): Promise<string[]> {
+  try {
+    const url = `./rest/api/1.0/users?filter=${encodeURIComponent(userinfo)}&permission.1=REPO_READ&permission.1.projectKey=${encodeURIComponent(
```

Comment:
`encodeURIComponent` is not needed, the request URI is automatically encoded as needed.